### PR TITLE
[feat]: [CDS-41933]: Gar triggers

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36882)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 28 Issues 
* Code issue 26 Code issues 
* Anti-pattern 2 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36882)

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java [hunk 1]
```diff
@@ -131,12 +131,11 @@ public GarDelegateRequest getGarDelegateRequest(GoogleArtifactRegistryConfig gar
         garArtifactConfig.getVersionRegex() != null ? garArtifactConfig.getVersionRegex().getValue() : "";
     String version = garArtifactConfig.getVersion() != null ? garArtifactConfig.getVersion().getValue() : "";
     if (StringUtils.isBlank(version) && StringUtils.isBlank(versionRegex)) {
-      versionRegex = ACCEPT_ALL_REGEX;
```
[the remark this answers is not in the document]
Comment (vivekkarnatiharness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java:134]: ACCEPT_ALL_REGEX("\\*") is not able to fetch all tags from getbuilds

**Review by puthrayaharness — APPROVED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger ti1

**Comment by vivekkarnatiharness** [comment]
trigger ti2

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 6 of 62 files at r1, 34 of 59 files at r2, 25 of 25 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36882)* status: :shipit: complete! ✓

**Comment by vivekkarnatiharness** [comment]
trigger ti2

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 62 files at r1, 4 of 59 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36882)* status: :shipit: complete! ✓

**Pull request merged**